src/network/network_server.cpp

```diff
@@ -1838,7 +1838,7 @@ static IntervalTimer<TimerGameCalendar> _network_daily({TimerGameCalendar::DAY,
 	if (!_network_server) return;
 
 	NetworkAdminUpdate(ADMIN_FREQUENCY_DAILY);
-	if ((TimerGameCalendar::date % 7) == 3) NetworkAdminUpdate(ADMIN_FREQUENCY_WEEKLY);
+	if (((int32_t)TimerGameCalendar::date % 7) == 3) NetworkAdminUpdate(ADMIN_FREQUENCY_WEEKLY);
```

Comment:
TODO: this should be fixed more properly, by adding `WEEK` and `QUARTER` similar as `DAY` in `TimerGameCalendar`, so a timer could be used here too.